From 168c8d65556ab18b05e33b5f4006d1184c30f8f7 Mon Sep 17 00:00:00 2001
From: Billy Donahue <billy.donahue@mongodb.com>
Date: Mon, 13 Aug 2018 12:44:11 -0400
Subject: [PATCH] SERVER-36611 warning suppression for gcc-8.2

---
 SConstruct | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SConstruct b/SConstruct
index 0df17d11778..31ca80632b9 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2146,6 +2146,10 @@ def doConfigure(myenv):
         # exceptionToStatus(). See https://bugs.llvm.org/show_bug.cgi?id=34804
         AddToCCFLAGSIfSupported(myenv, "-Wno-exceptions")
 
+        # These warnings begin in gcc-8.2 and we should get rid of these disables at some point.
+        AddToCCFLAGSIfSupported(env, '-Wno-format-truncation')
+        AddToCXXFLAGSIfSupported(env, '-Wno-class-memaccess')
+
 
         # Check if we can set "-Wnon-virtual-dtor" when "-Werror" is set. The only time we can't set it is on
         # clang 3.4, where a class with virtual function(s) and a non-virtual destructor throws a warning when
